[dtensor] account for empty list when turning to OpStrategy

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #115566
* #115565
* #115564
* #115297
* __->__ #115298

Trying to fix https://github.com/pytorch/pytorch/issues/115065

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225 @kiukchung @lucasllc

wanchaol committed:
[dtensor] account for empty list when turning to OpStrategy

Trying to fix https://github.com/pytorch/pytorch/issues/115065

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] account for empty list when turning to OpStrategy"


Trying to fix https://github.com/pytorch/pytorch/issues/115065

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu fduwjj wz337 tianyu-l wconstab yf225 kiukchung lucasllc

[ghstack-poisoned]

XilunWu reviewed (approved):
I'm curious of weather this empty list in the original issue comes from a op that has no return value or the return value is a list of optional objects?

wanchaol committed:
Update on "[dtensor] account for empty list when turning to OpStrategy"


Trying to fix https://github.com/pytorch/pytorch/issues/115065

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu fduwjj wz337 tianyu-l wconstab yf225 kiukchung lucasllc

[ghstack-poisoned]

wanchaol commented:
> I'm curious of weather this empty list in the original issue comes from a op that has no return value or the return value is a list of optional objects?

the issue mentioned that there's one argument that is a list[int] but it's a empty list when passing in, therefore erroring out

wanchaol commented:
@pytorchbot merge
